packages/xstate-wallet/src/workflows/advance-channel.ts

```diff
@@ -0,0 +1,85 @@
+import {AnyEventObject, MachineConfig, assign, spawn} from 'xstate';
+import {filter, map} from 'rxjs/operators';
+import {Store} from '../store';
+import {bigNumberify} from 'ethers/utils';
+import {connectToStore} from '../utils/workflow-utils';
+
+const WORKFLOW = 'advance-channel';
+/*
+Fully determined: true
+
+In the current wallet, the post-fund-setup version of advance-channel is responsible for
+storing state updates as they come in.
+In this spec, the store itself is responsible for that, so you can wait to spin up an
+advance-channel protocol once app funding is confirmed.
+
+Additionally, waiting until it's your turn isn't necessary once the channel is funded.
+An app should refrain from taking an app move until the entire post-fund round is supported,
+since their application updates are otherwise unenforcable.
+
+Therefore, we send on entry into the protocol.
+*/
+
+export interface Init {
+  channelId: string;
+  targetTurnNum: number; // should either be numParticipants-1 or 2*numParticipants-1
+}
+
+export const config: MachineConfig<Init, any, AnyEventObject> = {
+  key: WORKFLOW,
+  initial: 'sendState',
+  states: {
+    sendState: {
+      entry: 'spawnObserver',
+      invoke: {src: 'sendState'},
+      on: {ADVANCED: 'success'}
+    },
+    success: {type: 'final'}
+  }
+};
+
+export type Services = {
+  sendState(ctx: Init): Promise<void>;
+};
+
+const notifyWhenAdvanced = (store: Store, ctx: Init) => {
+  return store.channelUpdatedFeed(ctx.channelId).pipe(
+    filter(e => {
+      return !!e.supported && e.supported.turnNum.gte(ctx.targetTurnNum);
+    }),
+    map(() => 'ADVANCED')
+  );
+};
+
+const sendState = (store: Store) => async ({channelId, targetTurnNum}: Init) => {
```

Comment:
💡 This can be rewritten to use the `channelUpdatedFeed`:
- subscribe to the feed
- check the first entry you get
- if it looks ok, call `store.addState` and `take(1)` (so that the observable finishes)